generator/intermediate_data.hpp

```diff
@@ -215,222 +137,273 @@ class OSMElementCache
     return true;
   }
 
-  inline void SaveOffsets() { m_offsets.WriteAll(); }
-  inline void LoadOffsets() { m_offsets.ReadAll(); }
+  void LoadOffsets();
+
+protected:
+  FileReader m_fileReader;
+  IndexFileReader m_offsets;
+  std::string m_name;
+  std::vector<uint8_t> m_data;
+  bool m_preload = false;
 };
 
-/// Used to store all world nodes inside temporary index file.
-/// To find node by id, just calculate offset inside index file:
-/// offset_in_file = sizeof(LatLon) * node_ID
-class PointStorage
+class OSMElementCacheWriter
 {
-  size_t m_processedPoint = 0;
-
 public:
-  struct LatLon
-  {
-    int32_t lat;
-    int32_t lon;
-  };
-  static_assert(sizeof(LatLon) == 8, "Invalid structure size");
+  OSMElementCacheWriter(std::string const & name, bool preload = false);
```

Comment:
explicit? Because of the second param may have a default value.

Reply:
Yes, thanks.